src/Motor.Extensions.Hosting.RabbitMQ/BasicPropertiesExtensions.cs

```diff
@@ -27,12 +28,17 @@ static BasicPropertiesExtensions()
     }
 
     public static void Update<T>(this IBasicProperties self, MotorCloudEvent<byte[]> cloudEvent,
-        RabbitMQPublisherOptions<T> options)
+        RabbitMQPublisherOptions<T> options, CloudEventFormat format)
     {
         var messagePriority = cloudEvent.GetRabbitMQPriority() ?? options.DefaultPriority;
         if (messagePriority.HasValue)
             self.Priority = messagePriority.Value;
 
+        if (format == CloudEventFormat.Json)
+        {
+            return;
```

Comment:
Are you planning to add any headers or other indicators for consumers to be able to recognize the message format as Dapr?

Reply:
We thought about something like that but e.g. the RabbitMQ Dapr adapter also just sets the content type to `text/plain`.
According to the [docs](https://docs.dapr.io/reference/api/pubsub_api/#headers) we could probably set the content type to `application/cloudevents+json` but I'm not sure if this would have the expected result because we intended to publish in Dapr format without having to declare any kind of external binding to be able to preserve tracing headers and so on and so forth (which would not be possible if using external bindings).

Also I'm not sure if it is actually really necessary because right now the situation is more or less the same e.g. for JSON vs. Protobuf. If you don't know how to decode the payload you won't be able to do anything meaningful with it hence why even bothering? :smile: 
And in the future this would also allow to streamline how events are encoded no matter if sent with Kafka, RabbitMQ, NATS,...  because event metadata is always sent within  the payload and not protocol dependent?

Reply:
What about the content encoding? We currently set that to `gzip` if the message is compressed and services support consuming both compressed and uncompressed messages at the same time. So, this information is really needed to decide whether the payload needs to be uncompressed or not.
This won't be supported any more given the changes here or am I missing something?

Reply:
I guess the encoding can still be serialized with the cloud event even if not explicitly stated in the message header. It's just not as obvious anymore

Reply:
First staying with the protocol format nothing changes for existing services AFAIK. You can set whatever content type you want.
Second: the [CloudEvent spec](https://github.com/cloudevents/spec/blob/v1.0.1/json-format.md#23-examples) for the JSON envelope includes the `datacontenttype` header set in the JSON envelope which should be used e.g. for the `gzip` encoding and then you would but the binary data in the `data_base64` field no matter if it's GZIP compressed JSON, Protobuf, files, whatever?

Does this cover your questions?

Reply:
Yes, that covers my questions. I just didn't completely get the two formats on my first review pass through this PR.